### PR TITLE
[MI-2125]: Updated documentation for plugin phase-1

### PR DESCRIPTION
**Pull request by avas27JTG**
Updated documentation for plugin phase-1

### CODE REVIEW

**Review by Kshitij-Katiyar — APPROVED** [review]

File: docs/oauth_setup.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Setting up your AzureDevops for oAuth
+
+## Register your oAuth application on Azure DevOps
+  - Log in to [AzureDevops](https://app.vsaex.visualstudio.com).
+  - Click on the "Create new application" button in the bottom left corner.
+  - Fill in details like Company name, Application name.
+    ![Screenshot from 2022-07-29 13-24-47](https://user-images.githubusercontent.com/100013900/181712515-89efdb18-0f51-4194-b954-c0edb4188423.png)
+  - For the "Application website" put the link of your Mattermost server, for example: If the Mattermost server is running on `https://<your-mattermost-url>` then the value would be `https://<your-mattermost-url>` and "Authorization callback URL" will be `https://<your-mattermost-url>/plugins/mattermost-plugin-azure-devops/api/v1/oauth/complete`
+
+    ![Screenshot from 2022-07-29 13-25-38](https://user-images.githubusercontent.com/100013900/181712472-d4faec27-a61c-4565-9e27-fc7156241b17.png)
```
Comment (manojmalik20) [inline docs/oauth_setup.md:10]:
```suggestion
    ![Screenshot from 2022-07-29 13-25-38](https://user-images.githubusercontent.com/100013900/181712472-d4faec27-a61c-4565-9e27-fc7156241b17.png)

```

File: docs/oauth_setup.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  - Under "Authorized scopes" check "Work items(full)" and "Code(read)".
+
+    ![Screenshot from 2022-07-29 13-26-00](https://user-images.githubusercontent.com/100013900/181712419-3251837c-ea66-47b9-8b5b-7ae950816f2d.png)
+  - Click on "Create application".
```
Comment (manojmalik20) [inline docs/oauth_setup.md:14]:
```suggestion

  - Click on "Create application".
```

File: docs/plugin_setup.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Setting up your plugin configuration for oAuth
+
+  - Go to the Mattermost Azure Devops plugin configuration page on Mattermost as **System Console > Plugins > Mattermost Azure Devops plugin**.
```
Comment (manojmalik20) [inline docs/plugin_setup.md:3]:
```suggestion
  - Go to the Mattermost Azure DevOps plugin configuration page on Mattermost as **System Console > Plugins > Mattermost Azure DevOps plugin**.
```
We should replace Devops with `DevOps` everywhere.

File: README.md [hunk 4]
```diff
@@ -1,138 +1,87 @@
...
+## Overview
+
+This plugin integrates the services of Azure DevOps in Mattermost. For a stable production release, please download the latest version from the [Github Release](https://github.com/Brightscout/mattermost-plugin-azure-devops/releases) and follow the instructions to [install](#installation) and [configure](#configuration) the plugin.
+
+## Features
+
+This plugin contains the following features:
+- Right-hand sidebar (RHS) shows the list of linked projects and subscriptions created for each project with separate buttons to connect an account, link or unlink a project and add or delete a subscription.
+
+- Preview work item: A preview of the work item will be created when a work item URL is posted in a channel.
```
[the remark this answers is not in the document]
Comment (avas27JTG) [reply README.md:24]: no

File: README.md [hunk 5]
```diff
@@ -1,138 +1,87 @@
...
+
+- OAuth: A user can connect or disconnect to their Azure DevOps account using the slash command below or clicking on the "Connect Your Account" button in RHS.
```
[the remark this answers is not in the document]
Comment (avas27JTG) [reply README.md:26]: this point mentions `RHS` so we should first tell what is `RHS` here

File: docs/plugin_setup.md [hunk 6]
```diff
@@ -1,8 +1,8 @@
 # Setting up your plugin configuration for oAuth
 
-  - Go to the Mattermost Azure Devops plugin configuration page on Mattermost as **System Console > Plugins > Mattermost Azure Devops plugin**.
-  - On the Mattermost Azure Devops plugin configuration page, you need to configure the following:
-    - **Azure Devops API base URL**: Enter the base URL for Azure Devops API (`https://dev.azure.com`).
+  - Go to the Mattermost Azure DevOps plugin configuration page on Mattermost as **System Console > Plugins > Mattermost Azure Devops plugin**.
+  - On the Mattermost Azure DevOps plugin configuration page, you need to configure the following:
+    - **Azure Devops API base URL**: Enter the base URL for Azure DevOps API (`https://dev.azure.com`).
```
Comment (manojmalik20) [inline docs/plugin_setup.md:5]: Ideally, you should do this change here and in the `plugin.json` as well.
~Devops~ DevOps

**Review by manojmalik20 — APPROVED** [review]

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (manojmalik20) [reply README.md:24]: Then I think you should mention this in the documentation.

File: docs/plugin_setup.md (hunk 6 above)
Comment (avas27JTG) [reply docs/plugin_setup.md:5]: Done

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (avas27JTG) [reply README.md:24]: except `help` all features or commands requires account connection, so, should we mention that explicitly?

[the remark this answers is not in the document]
Comment (manojmalik20) [reply README.md:24]: No need to mention it for commands but I think we should mention it for this feature.

[the remark this answers is not in the document]
Comment (avas27JTG) [reply README.md:24]: updated here https://github.com/Brightscout/mattermost-plugin-azure-devops/pull/63/commits/d6b9f63ddbf54ccd9113d105724678dab99b2ce2

**Pull request merged**